internal/status/cache.go

```diff
@@ -0,0 +1,155 @@
+package status
+
+import (
+	"fmt"
+
+	projcontour "github.com/projectcontour/contour/apis/projectcontour/v1"
+	"github.com/projectcontour/contour/internal/k8s"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ConditionType string
+
+const ValidCondition ConditionType = "Valid"
+
+// ProxyUpdate holds status updates for a particular HTTPProxy object
+type ProxyUpdate struct {
```

Comment:
it'd be nice to have some tests for this type

Reply:
Agreed, I wanted to nail down if I was on the right track first. I'll make sure they're added before merging.